src/odb/include/odb/dbCompare.h

```diff
@@ -0,0 +1,909 @@
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbWire*>
+{
+  bool operator()(const odb::dbWire* lhs, const odb::dbWire* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbSWire*>
+{
+  bool operator()(const odb::dbSWire* lhs, const odb::dbSWire* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbTrackGrid*>
+{
+  bool operator()(const odb::dbTrackGrid* lhs,
+                  const odb::dbTrackGrid* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbObstruction*>
+{
+  bool operator()(const odb::dbObstruction* lhs,
+                  const odb::dbObstruction* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbBlockage*>
+{
+  bool operator()(const odb::dbBlockage* lhs, const odb::dbBlockage* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbCapNode*>
+{
+  bool operator()(const odb::dbCapNode* lhs, const odb::dbCapNode* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbRSeg*>
+{
+  bool operator()(const odb::dbRSeg* lhs, const odb::dbRSeg* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbCCSeg*>
+{
+  bool operator()(const odb::dbCCSeg* lhs, const odb::dbCCSeg* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbRow*>
+{
+  bool operator()(const odb::dbRow* lhs, const odb::dbRow* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbFill*>
+{
+  bool operator()(const odb::dbFill* lhs, const odb::dbFill* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbRegion*>
+{
+  bool operator()(const odb::dbRegion* lhs, const odb::dbRegion* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbLib*>
+{
+  bool operator()(const odb::dbLib* lhs, const odb::dbLib* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbSite*>
+{
+  bool operator()(const odb::dbSite* lhs, const odb::dbSite* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbMaster*>
+{
+  bool operator()(const odb::dbMaster* lhs, const odb::dbMaster* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbMTerm*>
+{
+  bool operator()(const odb::dbMTerm* lhs, const odb::dbMTerm* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
+
+template <>
+struct less<odb::dbMPin*>
+{
+  bool operator()(const odb::dbMPin* lhs, const odb::dbMPin* rhs) const
+  {
+    return odb::compare_by_id(lhs, rhs);
+  }
+};
```

Comment:
what happens if I want a set with MPins from multiple masters? In that case the mpin_table will be different and they will have non-unique ids (all compares where the allocation table isn't shared)

Reply:
If you look at compare_by_id you'll see it includes the parent's id into the comparison so that case should work.  Let me know if see otherwise.

Reply:
Ahh, I missed that. Thanks for clarifying